app/controllers/projects_controller.rb

```diff
@@ -38,14 +42,51 @@ def destroy
     @project.destroy
   end
 
+  # GET /projects?search=querystring
+  def search
+    begin
+      @projects = Project.get_project_by_querystring(params[:search]).order(created_at: :desc)
+    rescue ActiveRecord::RecordNotFound => e
+      render json: ErrorSerializer.new(e.message, Rack::Utils.status_code(:not_found)).serialized_json, status: :not_found
+    end
+    @pagy = pagy(@projects)
+    render json: ProjectSerializer.new(@projects, pagination_options).serialized_json
+  end
+
+  # GET /category/category_id/projects
+  def category_projects
+    begin
+      @projects = Project.category_projects(params[:category_id]).order(created_at: :desc)
+    rescue ActiveRecord::RecordNotFound => e
+      render json: ErrorSerializer.new(e.message, Rack::Utils.status_code(:not_found)).serialized_json, status: :not_found
+    end
+    @pagy = pagy(@projects)
+    render json: ProjectSerializer.new(@projects, pagination_options).serialized_json
+  end
+
+  # GET /user/user_id/projects
+  def user_projects
+    begin
+      @projects = Project.user_projects(params[:user_id]).order(created_at: :desc)
+    rescue ActiveRecord::RecordNotFound => e
+      render json: ErrorSerializer.new(e.message, Rack::Utils.status_code(:not_found)).serialized_json, status: :not_found
+    end
+    @pagy = pagy(@projects)
+    render json: ProjectSerializer.new(@projects, pagination_options).serialized_json
+  end
+
   private
     # Use callbacks to share common setup or constraints between actions.
     def set_project
-      @project = Project.find(params[:id])
+      begin
+        @project = Project.find(params[:id])
+      rescue ActiveRecord::RecordNotFound => e
+        render json: ErrorSerializer.new(e.message, Rack::Utils.status_code(:not_found)).serialized_json, status: :not_found
+      end
     end
 
     # Only allow a trusted parameter "white list" through.
     def project_params
-      params.require(:project).permit(:title, :description, :results, :user_id, :status_id)
+      params.require(:project).permit(:title, :description, :results, :status_id)
     end
 end
```

Comment:
Aggiungerei altri end-point, che possano restituirci i progetti in base a:
- se l'utente ne è admin
- se l'utente ne è membro

Li terrei volutamente separati da ```user_projects``` in modo da non dover fare controlli sulla presenza del parametro admin. Che dite?